### PR TITLE
Add OpRtnArg and OpRtnTag

### PR DESCRIPTION
**Pull request by tymm**
I couldn't find an expression that makes use of `opRtnArg` and `opRtnTag`.
Therefore I just made up two test cases:

`opRtnArg`:
```
litvec:
   0:   100
codevec:
   0:   liti 0,rslt
   1:   rtn/nxt arg[1]
```
`rtn/nxt arg[1]` corresponds to `OpRtnArg(next = true, arg = 1)` in Roscala.
`OpRtnArg(next = true, reg = 1)` copies the content of `state.ctxt.rslt` to `state.ctxt.argvec.elem(1)`.

---
`opRtnTag`:
```
litvec:
  0:   100
  1:   {Loc arg[1]}
codevec:
  0:   liti 0,rslt
  1:   rtn/tag/nxt
```
`rtn/tag/nxt` corresponds to `OpRtnTag(next = true, lit = 1)` in Roscala.
`OpRtnTag(next = true, lit = 1)` copies the content of `state.ctxt.rslt` to a location given in the `litvec`. In this case the location points to `state.ctxt.argvec.elem(1)`.

### CODE REVIEW

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala [hunk 1]
```diff
@@ -524,4 +524,61 @@ class TransitionSpec extends FlatSpec with Matchers {
...
+      * litvec:
+      *   0:   100
+      * codevec:
+      *   0:   liti 0,rslt
+      *   1:   rtn/nxt arg[1]
+      */
+    val start =
+      testState
+        .set(_ >> 'ctxt >> 'ctxt)(testState.ctxt)
+        .set(_ >> 'ctxt >> 'argvec)(Tuple(Seq(Ob.NIV, Ob.NIV, Ob.NIV)))
```
Comment (DoSOfRedRiver) [inline roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: Can you explain what this line is serving for?

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala [hunk 2]
```diff
@@ -524,4 +524,61 @@ class TransitionSpec extends FlatSpec with Matchers {
...
+        .update(_ >> 'code >> 'litvec)(_ => Tuple(Seq(Fixnum(100))))
```
Comment (DoSOfRedRiver) [inline roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: I think `set` could be used.

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala [hunk 3]
```diff
@@ -524,4 +524,61 @@ class TransitionSpec extends FlatSpec with Matchers {
...
+      *   1:   {Loc arg[2]}
+      * codevec:
+      *   0:   liti 0,rslt
+      *   1:   rtn/nxt
+      */
+    val start =
+      testState
+        .set(_ >> 'ctxt >> 'ctxt)(testState.ctxt)
+        .set(_ >> 'ctxt >> 'argvec)(Tuple(Seq(Ob.NIV, Ob.NIV, Ob.NIV)))
+        .update(_ >> 'code >> 'litvec)(_ =>
```
Comment (DoSOfRedRiver) [inline roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: See above.

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala [hunk 4]
```diff
@@ -524,4 +524,61 @@ class TransitionSpec extends FlatSpec with Matchers {
...
+          Tuple(Seq(Fixnum(100), ArgRegister(1))))
```
Comment (DoSOfRedRiver) [inline roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: In the description `Loc arg[2]` is specified, but here `ArgRegister(1)` is used. Did I miss something?

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala [hunk 5]
```diff
@@ -524,4 +524,61 @@ class TransitionSpec extends FlatSpec with Matchers {
...
+
+    val codevec = Seq(
+      OpIndLitToRslt(lit = 0),
+      OpRtnTag(next = true, lit = 1)
```
Comment (DoSOfRedRiver) [inline roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: Probably you forgot to update the opcode sequence.

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala (hunk 1 above)
Comment (tymm) [reply roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: `OpRtnArg(next = true, arg = 1)` writes to `state.ctxt.argvec.elem(1)` which needs to exist.

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala (hunk 2 above)
Comment (tymm) [reply roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: Done.

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala (hunk 3 above)
Comment (tymm) [reply roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: Done.

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala (hunk 4 above)
Comment (tymm) [reply roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: Good catch.

**Comment by tymm** [comment]
In this case it is just about tests to ensure that the transition function is doing the right thing.

**Pull request merged**